XUnitTestCommon/RestWrappers/RestRequests/Response.cs

```diff
@@ -21,11 +21,21 @@ public class Response : IResponse
 
     public class Response<T> : Response, IResponse<T>
     {
+        private T responseObject;
+
         IValidate<T> IResponse<T>.Validate => new Validate<T>(this);
 
-        public T GetResponseObject()
+        public T ResponseObject
         {
-            return JsonConvert.DeserializeObject<T>(Content);
+            get
+            {
+                if(Equals(responseObject, default(T)))
+                    responseObject = JsonConvert.DeserializeObject<T>(Content);
+
+                return responseObject;
```

Comment:
It's a kind of caching because when a response is deserialized once it won't be again for the same object. It was working like this: each time when you call the GetResponseObject() method - Deserialize. 
And about the ResponseObject instead of the GetReponseObject(), I discussed this with my .net colleagues and they told me that is better to use the object, not the method and there are no changes in the logic.